botocore/awsrequest.py

```diff
@@ -450,6 +471,92 @@ def prepare_body(self, data, files, json=None):
                 if 'Transfer-Encoding' in self.headers:
                     self.headers.pop('Transfer-Encoding')
 
+        if self.body and 'Content-Length' not in self.headers:
+            self.headers['Transfer-Encoding'] = 'chunked'
+
+
+class AWSResponse(object):
+
+    def __init__(self, url=None, status_code=None, headers=None, raw=None):
+        self.url = url
+        self.status_code = status_code
+        self.headers = HeadersDict(headers or {})
+        self.raw = raw
+
+        self._content = None
+
+    @property
+    def content(self):
+        """Content of the response, in bytes."""
+
+        if self._content is None:
+            # Read the contents.
+            try:
+                data_generator = self.raw.stream()
+            except AttributeError:
+                def stream():
```

Comment:
Is there a reason not to just have this be a separate method and always use it (vs. using `raw.stream()` if it's available).

Reply:
This is another case of 'it's what requests does'. I think the motivation here was that the Response object is generic and the underlying response could come from a different library other than urllib3 that may or may not have a stream method.

Reply:
In that case, I'd prefer to remove this if nothing's relying on it.  If we are going to keep it at least extract it out to a method instead of a nested function.